tests/unit/test_docs.py

```diff
@@ -53,6 +53,80 @@ def test_prog_name():
     assert output == HELLO_EXPECTED.replace("# hello", "# hello-world")
 
 
+def test_make_command_docs_invalid():
+    with pytest.raises(
+        MkDocsClickException, match="invalid is not a valid option style, which must be either 'plain' or 'table'."
+    ):
+        "\n".join(make_command_docs("hello", hello, style="invalid")).strip()
+
+
+@click.command()
+@click.option("-d", "--debug", help="Include debug output")
+@click.option("--choice", type=click.Choice(["foo", "bar"]), default="foo")
+@click.option("--date", type=click.DateTime(["%Y-%m-%d"]))
+@click.option("--range-a", type=click.FloatRange(0, 1), default=0)
+@click.option("--range-b", type=click.FloatRange(0))
+@click.option("--range-c", type=click.FloatRange(None, 1), default=0)
+def hello_table():
+    """Hello, world!"""
+
+
+HELLO_TABLE_EXPECTED = dedent(
+    """
+    # hello
+
+    Hello, world!
+
+    Usage:
+
+    ```
+    hello-table [OPTIONS]
+    ```
+
+    Options:
+
+    | Name | Type | Description | Default |
+    | ------ | ---- | ----------- | ------- |
+    | `-d`, `--debug` | text | Include debug output | _required_ |
+    | `--choice` | choice (`foo` &#x7C; `bar`) | No description given | `foo` |
+    | `--date` | datetime (`%Y-%m-%d`) | No description given | _required_ |
+    | `--range-a` | float range (between `0` and `1`) | No description given | `0` |
+    | `--range-b` | float range (`0` and above) | No description given | _required_ |
+    | `--range-c` | float range (`1` and below) | No description given | `0` |
```

Comment:
Is there an example in this test for the "`names_negeation`" logic? At first I thought `-d, --debug` could be, but it's lacking to  `/` so I'm thinking that's only `names`.

Reply:
Nope. It turns out that 100% coverage was a false positive because of a limitation of coverage (cf. nedbat/coveragepy#509). `secondary_opts` is used when the associated option is a [boolean flag](https://click.palletsprojects.com/en/7.x/options/?highlight=is_flag#boolean-flags), so I added a test accordingly.